### PR TITLE
Fixes null culture

### LINKED ISSUE #8722: Error during translation
Orchard.Localization.Services.GetCultureByName(string cultureName) throws an error when the cultureName is null
It happens during the translation phase

### PR DESCRIPTION
**Pull request by HermesSbicego-Laser**
fixes #8722

### CODE REVIEW

File: src/Orchard/Localization/Services/DefaultCultureManager.cs [hunk 1]
```diff
@@ -89,7 +89,7 @@ public class DefaultCultureManager : ICultureManager {
         public CultureRecord GetCultureByName(string cultureName) {
             var cultures = GetAllCulturesByName();
             CultureRecord result;
-            cultures.TryGetValue(cultureName, out result);
+            cultures.TryGetValue(cultureName ?? "", out result);
```
Comment (sebastienros) [inline src/Orchard/Localization/Services/DefaultCultureManager.cs]: If `""` doesn't have any meaning in terms of cultures and you just want a null record, then don't call `TryGetValue`, that will save two dictionary lookups.

Comment (sebastienros) [reply src/Orchard/Localization/Services/DefaultCultureManager.cs]:
```c#
if (string.IsNullOrEmpty(cultureName)) {
  return null;
}
... same code as before
```

Comment (sebastienros) [reply src/Orchard/Localization/Services/DefaultCultureManager.cs]: When will this cache be part of the dev branch?

Comment (HermesSbicego-Laser) [reply src/Orchard/Localization/Services/DefaultCultureManager.cs]: We don't have plans to merge 1.10.x into dev but we could talk about it in the next weeks with @BenedekFarkas: if the merge results to be too complex we could try to cherry pick the commit?

**Review by sebastienros — APPROVED** [review]

**Pull request merged**